somafm.el

```diff
@@ -106,6 +112,31 @@ Refresh channels list if necessary."
       (somafm--refresh-channels #'somafm--completing-read)
     (somafm--completing-read)))
 
+(defun somafm-add-current-song-to-favorites ()
+  "Add current song to favorites if it is not already starred.
+
+If the favorite file doesn't exist it will be created."
+  (interactive)
+  (if (not (somafm--favorite-p somafm-current-song))
+      (with-current-buffer (find-file-noselect somafm-favorites-file)
+        (save-excursion
+          (goto-char (point-max))
+          (insert (concat somafm-current-song (unless (looking-at "\n") "\n"))))
+        (save-buffer)
+        (message "Favorite saved in %s" somafm-favorites-file))
+    (message "Favorite already exists!")))
+
+(defun somafm-delete-current-song-from-favorites ()
+  (interactive)
+  (if (somafm--favorite-p somafm-current-song)
+      (with-current-buffer (find-file-noselect somafm-favorites-file)
+        (save-excursion                 ; in case the buffer is open
+          (goto-char (point-min))
+          (flush-lines (somafm--make-favorite-regexp somafm-current-song)))
+        (save-buffer)
+        (message "Favorite deleted from %s" somafm-favorites-file))
+    (message "Current song is not a favorite!")))
+
```

Comment:
We should give this interactive function a docstring for the user, similar to `somafm-add-current-song-to-favorites`